Missing values in r

rosemm opened the pull request.

pm0kjp commented on r_missing_values/r_missing_values.md:
```diff
@@ -0,0 +1,472 @@
+
+-->
+
+# @title
```
```suggestion
Missing Values in R
```
Unfortunately, using `@title` causes the left TOC to break, IDK why...

rosemm replied:
Ah, thank you! I keep forgetting that there's a good reason we don't use @title as a header...

pm0kjp commented on r_missing_values/r_missing_values.md:
```diff
@@ -0,0 +1,472 @@
+narrator: UK English Female
+title: Missing Values in R
+comment:  A practical demonstration of how missing values show up in R and how to deal with them. Note that this module does **not** cover statistical approaches for handling missing data, but instead focuses on the code you need to find, work with, and assign missing values in R.
+long_description: This is a beginners guide to handling missing values in R. It covers what `NA` values are and how to check for them, how to mark values as missing, how to work around missing values in your analysis, and how to filter your data to remove missingness. This module is appropriate for learners who feel comfortable with R basics and are ready to get into the realities of data analysi ...
```
```suggestion
long_description: This is a beginner's guide to handling missing values in R. It covers what `NA` values are and how to check for them, how to mark values as missing, how to work around missing values in your analysis, and how to filter your data to remove missingness. This module is appropriate for learners who feel comfortable with R basics and are ready to get into the realities of data analysis, like dealing with missing values.
```

pm0kjp commented on r_missing_values/r_missing_values.md:
```diff
@@ -0,0 +1,472 @@
+In R, missing values are marked with `NA`, which is short for "Not Available".
+
+<div class = "learnmore">
+Technically, there are actually four different kinds of `NA`, one for each of the different data types in R. To learn more, read about [missing values in the free online book R for Data Science](https://r4ds.had.co.nz/vectors.html?q=missing#missing-values-4)
```
```suggestion
Technically, there are actually four different kinds of `NA`, one for each of the different data types in R. To learn more, read about [missing values in the free online book R for Data Science](https://r4ds.had.co.nz/vectors.html?q=missing#missing-values-4).
```

pm0kjp commented on r_missing_values/r_missing_values.md:
```diff
@@ -0,0 +1,472 @@
+Let's break that down.
+
+We're using a `mutate` command to change the variable, and we're setting it to be conditional by using `if_else`.
+The `if_else` command begins with a conditional test, in this case `if_else(rating == -99, NA, rating)`, and for each value in `rating`, it will run the test and return either `TRUE` or `FALSE`.
```
```suggestion
The `if_else` command begins with a conditional test, in this case `rating == -99`.  Remember that the double equals sign is a comparison -- it's **asking** whether `rating` is equal to -99, while a single equals sign is a "setter", and it will try to **make** `rating` equal to -99.  For each value in `rating`, R will run the test `rating = -99` and return either `TRUE` or `FALSE`.
```

rosemm replied:
Couldn't batch this with the other suggestions because it was two conflicting suggestions for line 291, but I did implement it in a subsequent commit. Much better wording than what I had --- thank you!

pm0kjp commented on r_missing_values/r_missing_values.md:
```diff
@@ -0,0 +1,472 @@
+Let's break that down.
+
+We're using a `mutate` command to change the variable, and we're setting it to be conditional by using `if_else`.
+The `if_else` command begins with a conditional test, in this case `if_else(rating == -99, NA, rating)`, and for each value in `rating`, it will run the test and return either `TRUE` or `FALSE`.
```
```suggestion
The `if_else` command is what's called a ternary operator, and it has three parts:

 * a conditional test
 * a value to use if the test returns `TRUE`
 * a value to use if the test returns `FALSE`.
 
 Our `if_else` statement begins with a conditional test, in this case `if_else(rating == -99, NA, rating)`, and for each value in `rating`, it will run the test and return either `TRUE` or `FALSE`.
```

pm0kjp commented on r_missing_values/r_missing_values.md:
```diff
@@ -0,0 +1,472 @@
+
+Let's break that down.
+
+We're using a `mutate` command to change the variable, and we're setting it to be conditional by using `if_else`.
```
```suggestion
We're using a `mutate` command to change ("mutate") a variable (a column of data), and we're setting what the new value of this column should be by using a special conditional function, the `if_else` function.
```

pm0kjp reviewed (changes requested):
Hey @rosemm there was a big navigation error (@title usage breaks the TOC, IDK why) so I wanted to push this first batch of suggestions, but I do want to give this another once-over (and look at the code, which I haven't yet) before merging.

pm0kjp commented:
@rosemm I'm having a problem with line 118 (in the Binder version anyway).  Even when I put in your solution, I get this error:


<div class=" ace_lineWidgetContainer" style="color: rgb(0, 0, 0); font-family: Monaco, monospace, monospace; font-size: 12px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial; position: absolute; z-index: 3; top: -379px; left: 54.6035px; right: 15px;"><div class="GONVA-ECIBD GONVA-ECOT rstudio_chunk_output_unnamed_chunk_2" style="user-select: text; cursor: text; padding-bottom: 7px; position: relative; background-color: rgb(255, 255, 255); height: 257px;"><div class="GONVA-ECFBD" style="border: 1px solid rgba(0, 0, 0, 0.12); padding-top: 10px; overflow: hidden; background-color: rgb(255, 255, 255); height: 250px;"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACgAAAAkCAYAAAD7PHgWAAABi0lEQVR42u3XvU6DUBgG4E4u3oJjr4CBCxACowtOOhN+ll4CTiXMDnoNjRNxI65NY2PYcGtMWqCkxGo0+BNz5NMeg1ppoUdgOG/ypkPD16eH9gCtFg0NDU3zomlaW1XVw/TVSNstWGNxbPtfcIqiiKZpnriue+37/lMURahI4Rg4FmbALOIrZ1nWafohSVHYEmgCs4iuJJwa+Pab4nBhFswkuYJGmdP6V4MgeE6BRySB3TAMEcnCTKJAz/MQyZYGCoKwK4pi50edXu8MZTscXqGX17fSLQ0EEFojgHx4TKoH4hUcTyKUVwBOo1n1QAjHcfwqJADHk6AeYBYZTuPmAfFpht9aHN81C7gOrjZgHu4TVCMwDwf7oG2ff0NWClyB6/A8b49GN6jfH3whKwVi3Cye/8IxDHMgSdIx3pQHg8sPnONcVLcPLtucMS59e0eWZSs7HLC38/tqryTZzTmLS7sFwza57hL7F2Mky7L7GIfvZhoBXARQ2xgH0XV9r8RDUm5hJn0MpaGhqTHvbK0yKslszG8AAAAASUVORK5CYII=" width="40" height="36" class="GONVA-ECLBD" title="Show in New Window" style="position: absolute; top: 0px; padding: 5px; cursor: pointer; z-index: 25; right: 58px; width: 14px; height: 14px; opacity: 0.7;"><div class="GONVA-ECMAD" title="Clear Output" style="position: absolute; top: 2px; padding: 5px; cursor: pointer; z-index: 25; right: 12px; width: 11px !important; height: 10px !important;"><div class="GONVA-ECNAD" style="background-size: 100%; width: 11px; height: 10px; background-image: url(&quot;data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABYAAAAUCAYAAACJfM0wAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA4ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuNi1jMTM4IDc5LjE1OTgyNCwgMjAxNi8wOS8xNC0wMTowOTowMSAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDoxZjc0ZGZmNi00ZWEyLTRmNTQtOTkxNS1mN2E1OGQyODdmMzgiIHhtcE1NOkRvY3VtZW50SUQ9InhtcC5kaWQ6Qzg0QzdGM0NGQUIwMTFFNkI1MjA5MzExNzcwNUREOTYiIHhtcE1NOkluc3RhbmNlSUQ9InhtcC5paWQ6Qzg0QzdGM0JGQUIwMTFFNkI1MjA5MzExNzcwNUREOTYiIHhtcDpDcmVhdG9yVG9vbD0iQWRvYmUgUGhvdG9zaG9wIENDIDIwMTcgKE1hY2ludG9zaCkiPiA8eG1wTU06RGVyaXZlZEZyb20gc3RSZWY6aW5zdGFuY2VJRD0ieG1wLmlpZDo2YmU2Yjg2ZC1jNTQ1LTQzZTEtOGYxNi03MTg1NWVkZTVlNGYiIHN0UmVmOmRvY3VtZW50SUQ9ImFkb2JlOmRvY2lkOnBob3Rvc2hvcDpkNTJiMzQxOC00MzE3LTExN2EtYjhmOS1kMTcyMzliNGI4YjAiLz4gPC9yZGY6RGVzY3JpcHRpb24+IDwvcmRmOlJERj4gPC94OnhtcG1ldGE+IDw/eHBhY2tldCBlbmQ9InIiPz5gxgbbAAAA00lEQVR42rTUUQrDIAwAUCvtyuh+R4+wa20fO1V3sR1B9quUwsBFsCAh0RRnIB+F+JCYtPPeqxahVaNoBvfpx+V6q8Ls51288Qg5CaxQc5K2IhTeIZ8FfIo1Dwqn4AHyDDln8B2dY+0ggR3kC9IweIqaWOukU2EZnEItBXTpghBTgSGVQyVTkbZlSW5evGntgvjazcOtMIVpEcEYXVBbsrg+gDqi5yyuD6Dcg5I4BX8hVwal8DWeEc3xGP98TvC4Ad3wHPfMgW0vFsy5Km7eP+MnwADdyU/tIwjohwAAAABJRU5ErkJggg==&quot;);"></div></div><div class="GONVA-ECCBD" title="Expand/Collapse Output" style="position: absolute; top: 2px; padding: 5px; cursor: pointer; z-index: 25; width: 11px !important; height: 10px !important; right: 35px; transition: transform 400ms ease 0s, -webkit-transform 400ms ease 0s;"><div class="GONVA-ECDBD" style="background-size: 100%; width: 11px; height: 10px; background-image: url(&quot;data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABYAAAAUCAYAAACJfM0wAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA4ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuNi1jMTM4IDc5LjE1OTgyNCwgMjAxNi8wOS8xNC0wMTowOTowMSAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDoxZjc0ZGZmNi00ZWEyLTRmNTQtOTkxNS1mN2E1OGQyODdmMzgiIHhtcE1NOkRvY3VtZW50SUQ9InhtcC5kaWQ6Qzg0QzdGMzRGQUIwMTFFNkI1MjA5MzExNzcwNUREOTYiIHhtcE1NOkluc3RhbmNlSUQ9InhtcC5paWQ6Qzg0QzdGMzNGQUIwMTFFNkI1MjA5MzExNzcwNUREOTYiIHhtcDpDcmVhdG9yVG9vbD0iQWRvYmUgUGhvdG9zaG9wIENDIDIwMTcgKE1hY2ludG9zaCkiPiA8eG1wTU06RGVyaXZlZEZyb20gc3RSZWY6aW5zdGFuY2VJRD0ieG1wLmlpZDo2YmU2Yjg2ZC1jNTQ1LTQzZTEtOGYxNi03MTg1NWVkZTVlNGYiIHN0UmVmOmRvY3VtZW50SUQ9ImFkb2JlOmRvY2lkOnBob3Rvc2hvcDpkNTJiMzQxOC00MzE3LTExN2EtYjhmOS1kMTcyMzliNGI4YjAiLz4gPC9yZGY6RGVzY3JpcHRpb24+IDwvcmRmOlJERj4gPC94OnhtcG1ldGE+IDw/eHBhY2tldCBlbmQ9InIiPz63LKxbAAAAwUlEQVR42mL8//8/Ay0AEwONAM0MZkEXSE1NBdOPHr1El2qC0nVycuIYBs2ePZssFzchs7FYSnxQ7Ny5CYxhhu7atXkLCMMMB7kQ3ZVEu9jd3Q9uKJA6BcIww2FyINdj8wETCYbCAIbhRLsY6AJchmIYjhb+uA0mZChShsJrOEZyAyalOmyGIXm7DsRnZGQEGw5Uf4qsDIJmKN5wJcpgkIHIhiInN5gYvnKG7OTm5uY7UpMblvDEazjjaEEPAwABBgBxs6XcbDPKhQAAAABJRU5ErkJggg==&quot;);"></div></div><div class="GONVA-ECMBD" style="position: relative; transition: opacity 400ms ease 0s; margin: 10px; opacity: 1;"><div><div data-ordinal="5"><div data-pagedtable="false"><div class="pagedtable pagedtable-not-empty" style="overflow: auto; padding-left: 8px; padding-right: 8px; font-family: &quot;Lucida Sans&quot;, &quot;DejaVu Sans&quot;, &quot;Lucida Grande&quot;, &quot;Segoe UI&quot;, Verdana, Helvetica, sans-serif; opacity: 1;"><div><div class="pagedtable-metacell" style="padding-left: 0px; padding-bottom: 10px; display: flex; flex-direction: row;"><div style="display: inline-block; border: 1px solid rgb(233, 233, 233); border-radius: 3px; padding: 4px; margin-left: 0px; margin-right: 10px; font-size: 8pt !important;"><span class="pagedtable-metafield" style="color: rgb(153, 153, 153); padding-right: 4px;">A tibble:</span><span class="pagedtable-metaval">6 × 11</span></div></div></div>

 
--


<div class="pagedtable-footer" style="padding-top: 9px; padding-bottom: 5px; max-height: 18px; overflow: hidden; border-top: 1px solid rgba(0, 0, 0, 0.12);"><div class="pagedtable-info" title="4 rows | 1-8 of 11 columns" style="overflow: hidden; max-height: 20px; color: rgb(153, 153, 153); white-space: nowrap; text-overflow: ellipsis;">4 rows | 1-8 of 11 columns</div></div></div></div></div></div></div></div></div></div></div></div><div class=" ace_lineWidgetContainer" style="color: rgb(0, 0, 0); font-family: Monaco, monospace, monospace; font-size: 12px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial; position: absolute; z-index: 3; top: 95px; left: 54.6035px; right: 15px;"><div class="GONVA-ECIBD GONVA-ECOT rstudio_chunk_output_unnamed_chunk_13" style="user-select: text; cursor: text; padding-bottom: 7px; position: relative; background-color: rgb(255, 255, 255); height: 215px;"><div class="GONVA-ECFBD" style="border: 1px solid rgba(0, 0, 0, 0.12); padding-top: 10px; overflow: hidden; background-color: rgb(255, 255, 255); height: 208px;"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACgAAAAkCAYAAAD7PHgWAAABi0lEQVR42u3XvU6DUBgG4E4u3oJjr4CBCxACowtOOhN+ll4CTiXMDnoNjRNxI65NY2PYcGtMWqCkxGo0+BNz5NMeg1ppoUdgOG/ypkPD16eH9gCtFg0NDU3zomlaW1XVw/TVSNstWGNxbPtfcIqiiKZpnriue+37/lMURahI4Rg4FmbALOIrZ1nWafohSVHYEmgCs4iuJJwa+Pab4nBhFswkuYJGmdP6V4MgeE6BRySB3TAMEcnCTKJAz/MQyZYGCoKwK4pi50edXu8MZTscXqGX17fSLQ0EEFojgHx4TKoH4hUcTyKUVwBOo1n1QAjHcfwqJADHk6AeYBYZTuPmAfFpht9aHN81C7gOrjZgHu4TVCMwDwf7oG2ff0NWClyB6/A8b49GN6jfH3whKwVi3Cye/8IxDHMgSdIx3pQHg8sPnONcVLcPLtucMS59e0eWZSs7HLC38/tqryTZzTmLS7sFwza57hL7F2Mky7L7GIfvZhoBXARQ2xgH0XV9r8RDUm5hJn0MpaGhqTHvbK0yKslszG8AAAAASUVORK5CYII=" width="40" height="36" class="GONVA-ECLBD" title="Show in New Window" style="position: absolute; top: 0px; padding: 5px; cursor: pointer; z-index: 25; right: 58px; width: 14px; height: 14px; opacity: 0.7;"><div class="GONVA-ECMAD" title="Clear Output" style="position: absolute; top: 2px; padding: 5px; cursor: pointer; z-index: 25; right: 12px; width: 11px !important; height: 10px !important;"><div class="GONVA-ECNAD" style="background-size: 100%; width: 11px; height: 10px; background-image: url(&quot;data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABYAAAAUCAYAAACJfM0wAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA4ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuNi1jMTM4IDc5LjE1OTgyNCwgMjAxNi8wOS8xNC0wMTowOTowMSAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDoxZjc0ZGZmNi00ZWEyLTRmNTQtOTkxNS1mN2E1OGQyODdmMzgiIHhtcE1NOkRvY3VtZW50SUQ9InhtcC5kaWQ6Qzg0QzdGM0NGQUIwMTFFNkI1MjA5MzExNzcwNUREOTYiIHhtcE1NOkluc3RhbmNlSUQ9InhtcC5paWQ6Qzg0QzdGM0JGQUIwMTFFNkI1MjA5MzExNzcwNUREOTYiIHhtcDpDcmVhdG9yVG9vbD0iQWRvYmUgUGhvdG9zaG9wIENDIDIwMTcgKE1hY2ludG9zaCkiPiA8eG1wTU06RGVyaXZlZEZyb20gc3RSZWY6aW5zdGFuY2VJRD0ieG1wLmlpZDo2YmU2Yjg2ZC1jNTQ1LTQzZTEtOGYxNi03MTg1NWVkZTVlNGYiIHN0UmVmOmRvY3VtZW50SUQ9ImFkb2JlOmRvY2lkOnBob3Rvc2hvcDpkNTJiMzQxOC00MzE3LTExN2EtYjhmOS1kMTcyMzliNGI4YjAiLz4gPC9yZGY6RGVzY3JpcHRpb24+IDwvcmRmOlJERj4gPC94OnhtcG1ldGE+IDw/eHBhY2tldCBlbmQ9InIiPz5gxgbbAAAA00lEQVR42rTUUQrDIAwAUCvtyuh+R4+wa20fO1V3sR1B9quUwsBFsCAh0RRnIB+F+JCYtPPeqxahVaNoBvfpx+V6q8Ls51288Qg5CaxQc5K2IhTeIZ8FfIo1Dwqn4AHyDDln8B2dY+0ggR3kC9IweIqaWOukU2EZnEItBXTpghBTgSGVQyVTkbZlSW5evGntgvjazcOtMIVpEcEYXVBbsrg+gDqi5yyuD6Dcg5I4BX8hVwal8DWeEc3xGP98TvC4Ad3wHPfMgW0vFsy5Km7eP+MnwADdyU/tIwjohwAAAABJRU5ErkJggg==&quot;);"></div></div><div class="GONVA-ECCBD" title="Expand/Collapse Output" style="position: absolute; top: 2px; padding: 5px; cursor: pointer; z-index: 25; width: 11px !important; height: 10px !important; right: 35px; transition: transform 400ms ease 0s, -webkit-transform 400ms ease 0s;"><div class="GONVA-ECDBD" style="background-size: 100%; width: 11px; height: 10px; background-image: url(&quot;data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABYAAAAUCAYAAACJfM0wAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA4ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuNi1jMTM4IDc5LjE1OTgyNCwgMjAxNi8wOS8xNC0wMTowOTowMSAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDoxZjc0ZGZmNi00ZWEyLTRmNTQtOTkxNS1mN2E1OGQyODdmMzgiIHhtcE1NOkRvY3VtZW50SUQ9InhtcC5kaWQ6Qzg0QzdGMzRGQUIwMTFFNkI1MjA5MzExNzcwNUREOTYiIHhtcE1NOkluc3RhbmNlSUQ9InhtcC5paWQ6Qzg0QzdGMzNGQUIwMTFFNkI1MjA5MzExNzcwNUREOTYiIHhtcDpDcmVhdG9yVG9vbD0iQWRvYmUgUGhvdG9zaG9wIENDIDIwMTcgKE1hY2ludG9zaCkiPiA8eG1wTU06RGVyaXZlZEZyb20gc3RSZWY6aW5zdGFuY2VJRD0ieG1wLmlpZDo2YmU2Yjg2ZC1jNTQ1LTQzZTEtOGYxNi03MTg1NWVkZTVlNGYiIHN0UmVmOmRvY3VtZW50SUQ9ImFkb2JlOmRvY2lkOnBob3Rvc2hvcDpkNTJiMzQxOC00MzE3LTExN2EtYjhmOS1kMTcyMzliNGI4YjAiLz4gPC9yZGY6RGVzY3JpcHRpb24+IDwvcmRmOlJERj4gPC94OnhtcG1ldGE+IDw/eHBhY2tldCBlbmQ9InIiPz63LKxbAAAAwUlEQVR42mL8//8/Ay0AEwONAM0MZkEXSE1NBdOPHr1El2qC0nVycuIYBs2ePZssFzchs7FYSnxQ7Ny5CYxhhu7atXkLCMMMB7kQ3ZVEu9jd3Q9uKJA6BcIww2FyINdj8wETCYbCAIbhRLsY6AJchmIYjhb+uA0mZChShsJrOEZyAyalOmyGIXm7DsRnZGQEGw5Uf4qsDIJmKN5wJcpgkIHIhiInN5gYvnKG7OTm5uY7UpMblvDEazjjaEEPAwABBgBxs6XcbDPKhQAAAABJRU5ErkJggg==&quot;);"></div></div><div class="GONVA-ECMBD" style="position: relative; transition: opacity 400ms ease 0s; margin: 10px; opacity: 1;"><div><pre data-ordinal="1" style="font-family: Monaco, monospace, monospace; font-size: 9pt !important; line-height: 1.45; margin-bottom: 0px; margin-top: 0px; white-space: pre-wrap;"><span class="xtermBold xtermColor3" style="font-weight: bold; color: rgb(196, 160, 0) !important;">Error</span><span class="xtermBold" style="font-weight: bold;"> in </span><span class="xtermBold xtermColor232" style="font-weight: bold; color: rgb(8, 8, 8) !important;">`mutate()`</span><span class="xtermBold" style="font-weight: bold;">:</span><span>
</span><span class="xtermColor3" style="color: rgb(196, 160, 0) !important;">!</span><span class="xtermColor232" style="color: rgb(8, 8, 8) !important;"> Problem while computing `sleep_total = if_else(sleep_total &gt; 18, NA, sleep_total)`.</span><span>
</span><span class="xtermBold" style="font-weight: bold;">Caused by error in </span><span class="xtermBold xtermColor232" style="font-weight: bold; color: rgb(8, 8, 8) !important;">`if_else()`</span><span class="xtermBold" style="font-weight: bold;">:</span><span>
</span><span class="xtermColor3" style="color: rgb(196, 160, 0) !important;">!</span><span class="xtermColor232" style="color: rgb(8, 8, 8) !important;"> `false` must be a logical vector, not a double vector.</span><span>
Backtrace:
</span><span class="xtermColor8" style="color: rgb(85, 87, 83) !important;"> 1. </span><span class="xtermBold" style="font-weight: bold;">dplyr</span><span>::mutate(...)
</span><span class="xtermColor8" style="color: rgb(85, 87, 83) !important;"> 6. </span><span>dplyr::if_else(sleep_total &gt; 18, NA, sleep_total)
</span></pre><div class="GONVA-ECHGD ace_console_error" style="position: relative; overflow: hidden; margin: 5px 5px 5px 0px; padding: 5px 0px 5px 10px; white-space: pre-line; background-color: rgb(230, 230, 230);"><div class="GONVA-ECIGD" style="top: 5px; float: right; margin-right: 10px; margin-left: 5px; margin-bottom: 15px; line-height: 9.6px; position: relative; user-select: none;"><div><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABIAAAASCAYAAABWzo5XAAAAaElEQVR42mNgwAXWv8sEY4rA2vd+DOve/wFjEJsssOGjOdCA/ygYJEYS2PhRBcMQGAbJEQVWfRbFaQgMg9TgBZufcRE0BIZBarG75D8z0YbAXQbUgwFINQSGCQKyNY4aRLoFw9iggQYAE6xnKErSS9EAAAAASUVORK5CYII=" width="18" height="18" class="GONVA-ECJGD show_traceback_image" alt="" aria-hidden="true" style="cursor: pointer; width: 9px !important; height: 9px !important;"> <a class="GONVA-ECKGD show_traceback_text" style="font-family: &quot;Lucida Sans&quot;, &quot;DejaVu Sans&quot;, &quot;Lucida Grande&quot;, &quot;Segoe UI&quot;, Verdana, Helvetica, sans-serif, serif; font-size: 9.6px; opacity: 0.5; cursor: pointer;">Show Traceback</a></div><div></div></div><div class="GONVA-ECLGD ace_constant" style="display: inline; overflow-wrap: break-word; word-break: break-word; color: rgb(197, 6, 11);"><span>Error in mutate(msleep_clean, sleep_total = if_else(sleep_total &gt; 18, :
</span><span class="xtermBold" style="font-weight: bold;">Caused by error in </span><span class="xtermBold xtermColor232" style="font-weight: bold; color: rgb(8, 8, 8) !important;">`if_else()`</span><span class="xtermBold" style="font-weight: bold;">:</span><span>
</span><span class="xtermColor3" style="color: rgb(196, 160, 0) !important;">!</span><span class="xtermColor232" style="color: rgb(8, 8, 8) !important;"> `false` must be a logical vector, not a double vector.</span></div></div></div></div></div></div></div>

pm0kjp commented:
@rosemm Just a thought, when you say in the code on line 196, 

> It produces the scatterplot we asked for, but you'll also see a warning saying "Removed 27 rows containing missing values (geom_point)" to remind you that it's ignoring any observations that are missing for either of the two variables we're plotting

it might be a good idea to tell folks that in R Markdown, that error is found in the "R Console" box below the code chunk as well as in the Console below... otherwise they may just say "huh?  I see a graph with no error".

rosemm commented:
Thanks, @pm0kjp! Excellent suggestions. I implemented them all. For the if_else error, that was my bad -- it should be ifelse, not if_else, and is now corrected in the module and binder instance.

pm0kjp reviewed (approved):
LGTM!
